src/pages/Chat/steps/NameAgeInput.tsx

```diff
@@ -0,0 +1,116 @@
+import { Container, StateUpdater } from 'constate'
+import * as React from 'react'
+import {
+  UserResponse,
+  UserTextInput,
+} from '../../../components/userInput/UserResponse'
+import { ChatContainer } from '../state'
+
+interface State {
+  firstName: string
+  lastName: string
+  age: string
+  hasFocused: boolean
+}
+
+interface Actions {
+  setFirstName: (firstName: string) => StateUpdater<State>
+  setLastName: (lastName: string) => StateUpdater<State>
+  setAge: (age: string) => StateUpdater<State>
+  setFocused: () => StateUpdater<State>
+}
+
+interface Props {
+  onSubmit: (state?: State) => void
+}
+
+const isDone = (state: State) =>
+  state.firstName.length > 0 &&
+  state.lastName.length > 0 &&
+  state.age.length >= 2
+
+export const NameAgeInput: React.SFC<Props> = ({ onSubmit }) => (
+  <ChatContainer>
+    {({ setStep1 }) => (
+      <Container<State, Actions>
+        initialState={{
+          firstName: '',
+          lastName: '',
+          age: '',
+          hasFocused: false,
+        }}
+        actions={{
+          setFirstName: (firstName) => (_) => ({ firstName }),
+          setLastName: (lastName) => (_) => ({ lastName }),
+          setAge: (age) => (_) => {
+            if (/^\d{0,3}$/.test(age)) {
+              return { age }
+            }
+            return {}
+          },
+          setFocused: () => (_) => ({ hasFocused: true }),
+        }}
+      >
+        {(state) => (
+          <UserResponse>
```

Comment:
I think `Formik` could be worthwhile to try out here 🙂

Reply:
i'll take a look!